crates/rbuilder/src/live_builder/base_config.rs

```diff
@@ -590,29 +608,42 @@ pub fn coinbase_signer_from_secret_key(secret_key: &str) -> eyre::Result<Signer>
     Ok(Signer::try_from_secret(secret_key)?)
 }
 
-fn get_signing_domain(chain: Chain, cl_network_config_dir: Option<String>) -> eyre::Result<B256> {
-    let network = named_chain_to_network(chain)
-        .or_else(|| cl_network_config_dir.map(Network::Custom))
-        .ok_or_else(|| {
-            eyre::eyre!("Unknown chain. If network is not named, specify cl_network_config_dir")
-        })?;
+fn get_signing_domain(chain: Chain, client: Client) -> eyre::Result<B256> {
+    let cl_context = match chain.kind() {
+        ChainKind::Named(NamedChain::Mainnet) => ContextEth::for_mainnet(),
+        ChainKind::Named(NamedChain::Sepolia) => ContextEth::for_sepolia(),
+        ChainKind::Named(NamedChain::Goerli) => ContextEth::for_goerli(),
+        ChainKind::Named(NamedChain::Holesky) => ContextEth::for_holesky(),
+        _ => {
+            let rt = Runtime::new()?;
+            let spec = rt.block_on(client.get_spec())?;
```

Comment:
Maybe Handle::block_on is more idiomatic and can be wrapped in task::block_in_place? https://docs.rs/tokio/latest/tokio/task/fn.block_in_place.html#examples

Reply:
Cannot make it work. I think there is some nuance there that makes these two functions different.